bin/rust/rust_belt/lifetime_logic.gh

```diff
@@ -76,24 +89,24 @@ lemma void full_borrow_implies(lifetime_t k, predicate() P, predicate() P1);
     ensures full_borrow(k, P1) &*& is_implies(f, P, P1) &*& is_implies(f1, P1, P);
 
 // Non-atomic persistent borrow
-predicate nonatomic_borrow(lifetime_t k, thread_id_t t, predicate() P);
+predicate nonatomic_borrow(lifetime_t k, thread_id_t t, mask_t m, predicate() P);
 
-lemma void full_borrow_into_nonatomic_borrow(lifetime_t k, thread_id_t t, predicate() P);
+lemma void full_borrow_into_nonatomic_borrow(lifetime_t k, thread_id_t t, predicate() P, mask_t m);
```

Comment:
I'm pretty sure this mask must not be empty!! Same bug as before?

Reply:
Yes indeed